classes/notification/NotificationDto.php

```diff
@@ -10,6 +10,7 @@ public function __construct(
         public int $created = 0,
         public int $duration = 0,
         public bool $alert = false,
+        public array $attributes = [],
```

Comment:
Instead of an untyped array, could we make this typed? I think everywhere else we have data in an untyped array it becomes more difficult to work with. Some options here include

A) Create subclasses of NotificationDto for each type that needs attributes. For instance `MissionNotificationDto`  that adds on the `mission_id` and `mission_url` fields to the class, internally it gets serialized to/from a JSON string for storage in the database, but anywhere in PHP you work with it, it's typed

B) Create just a subclass for the attributes itself, like `NotificationMissionAttributes`. Make the `attributes` field on this class a union of `null | NotificationMissionAttributes | NotificationSomeOtherAttributes`. So that way when accessing it, we can check `instanceof NotificationMissionAttributes` and then know exactly what we're dealing with and what fields are set